File: phalcon/library/tencent/Im.php
<?php
namespace Library\Tencent;

use Config\Tencent;
use Library\Curl;

/* 即时通信 */
class Im extends Signature {

  /* 请求地址 */
  static function GetURL(string $apiUrl) {
    $cfg = Tencent::TRTC();
    $userSig = self::UserSig($cfg['UserID']);
    $random = time();
    return 'https://console.tim.qq.com/v4/'.$apiUrl.'?sdkappid='.$cfg['SDKAppID'].'&identifier='.$cfg['UserID'].'&usersig='.$userSig.'&random='.$random.'&contenttype=json';
  }

  /* 群组-列表 */
  static function GroupList() {
    $url = self::GetURL('group_open_http_svc/get_appid_group_list');
    return Curl::PostJson($url, []);
  }
  /* 群组-创建 */
  static function GroupCreate(array $data) {
    $url = self::GetURL('group_open_http_svc/create_group');
    return Curl::PostJson($url, $data);
  }
  /* 群组-解散 */
  static function GroupDestroy(array $data) {
    $url = self::GetURL('group_open_http_svc/destroy_group');
    return Curl::PostJson($url, $data);
  }

}

File: java/src/main/java/webmis/library/tencent/Im.java
package webmis.library.tencent;

import java.util.HashMap;

import com.alibaba.fastjson.JSONObject;

import webmis.config.Tencent;
import webmis.library.Curl;
import webmis.util.Util;

/* 即时通信 */
public class Im extends Signature {

  /* 请求地址 */
  public static String GetURL(String apiUrl) {
    HashMap<String, Object> cfg = Tencent.TRTC();
    String userSig = UserSig(cfg.get("UserID"));
    String random = String.valueOf(Util.Time());
    return "https://console.tim.qq.com/v4/"+apiUrl+"?sdkappid="+cfg.get("SDKAppID")+"&identifier="+cfg.get("UserID")+"&usersig="+userSig+"&random="+random+"&contenttype=json";
  }

  /* 群组-列表 */
  public static JSONObject GroupList() {
    String url = GetURL("group_open_http_svc/get_appid_group_list");
    return Curl.Request(url, "[]", "POST");
  }
  /* 群组-创建 */
  public static JSONObject GroupCreate(JSONObject data) {
    String url = GetURL("group_open_http_svc/create_group");
    return Curl.Request(url, data.toString(), "POST");
  }
  /* 群组-解散 */
  public static JSONObject GroupDestroy(JSONObject data) {
    String url = GetURL("group_open_http_svc/destroy_group");
    return Curl.Request(url, data.toString(), "POST");
  }
  
}
